purchases/src/main/kotlin/com/revenuecat/purchases/common/CustomerInfoFactory.kt

```diff
@@ -78,6 +82,17 @@ internal object CustomerInfoFactory {
                 Iso8601Utils.parse(it) ?: null
             }
 
+        val virtualCurrenciesObject = subscriber.optJSONObject(
+            CustomerInfoResponseJsonKeys.VIRTUAL_CURRENCIES,
+        ) ?: JSONObject()
+
+        val virtualCurrencies = buildMap<String, VirtualCurrencyInfo> {
+            virtualCurrenciesObject.keys().forEach { currencyId ->
+                val currencyJson = virtualCurrenciesObject.getJSONObject(currencyId)
+                put(currencyId, VirtualCurrencyInfo.fromJson(currencyJson))
+            }
+        }
```

Comment:
I'm wondering, since this is still experimental, should we be defensive here and catch errors parsing this data? In case we want to change the backend response for any reason while it's experimental.

Reply:
Works for me! I'll modify it to default to an empty map if any parsing errors occur